performance/admin.py

```diff
@@ -0,0 +1,404 @@
+from django.contrib import admin
+from io import StringIO
+from performance import models
+from django_q.tasks import async_task
+
+from frictionless import validate
+
+import os
+import csv
+
+VALID_REPORT_TYPES = ['Provincial Institutions Oversight Performance  Report',
+                      'National Institutions Oversight Performance  Report']
+
+
+def generate_import_report(report_type_validated, frictionless_report, not_matching_departments):
+    report = ""
+    if not report_type_validated:
+        report += "Report preamble must be for one of " + os.linesep
+        for report_type in VALID_REPORT_TYPES:
+            report += f"* {report_type} {os.linesep}"
+
+    if frictionless_report:
+        if not frictionless_report.valid:
+            for error in frictionless_report.tasks[0].errors:
+                report += f"* {error.message} {os.linesep}"
+
+    if len(not_matching_departments) > 0:
+        report += "Department names that could not be matched on import : " + os.linesep
+        for department in not_matching_departments:
+            report += f"* {department} {os.linesep}"
+
+    return report
+
+
+def save_imported_indicators(parsed_data, obj_id):
+    report_departments = set([x['Institution'] for x in parsed_data])
+    num_imported = 0
+    total_record_count = len(parsed_data)
+    not_matching_departments = []
+
+    for department in report_departments:
+        department_obj = models.Department.objects.filter(name=department).first()
+        models.Indicator.objects.filter(department=department_obj).delete()
+
+    for indicator_data in parsed_data:
+        department_name = indicator_data['Institution']
+        department_obj = models.Department.objects.filter(name=department_name).first()
```

Comment:
department lookup here needs to consider government, sphere, etc as above

Reply:
done